modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C

```diff
@@ -66,13 +72,23 @@ INSFVTKEDSourceSink::INSFVTKEDSourceSink(const InputParameters & params)
     _C2_eps(getParam<Real>("C2_eps")),
     _C_mu(getParam<Real>("C_mu")),
     _C_pl(getParam<Real>("C_pl")),
-    _newton_solve(getParam<bool>("newton_solve"))
+    _newton_solve(getParam<bool>("newton_solve")),
+    _temperature(params.isParamValid(NS::temperature) ? &(getFunctor<ADReal>(NS::temperature)) : nullptr),
+    _alpha(params.isParamValid("alpha_name") ? &(getFunctor<ADReal>("alpha_name")) : nullptr),
+    _Pr_t(params.isParamValid(NS::turbulent_Prandtl) ? &(getFunctor<ADReal>(NS::turbulent_Prandtl)) : nullptr),
+    _gravity(params.isParamValid("gravity") ? &getParam<RealVectorValue>("gravity") : nullptr)
 {
   if (_dim >= 2 && !_v_var)
     paramError("v", "In two or more dimensions, the v velocity must be supplied!");
 
   if (_dim >= 3 && !_w_var)
     paramError("w", "In three or more dimensions, the w velocity must be supplied!");
+
+  if(_temperature && _alpha == 0)
+    paramError("alpha", "The thermal expansion coefficient should be >0.0 for themal bouyancy production correction.");
+
+  if(_temperature && !(_gravity))
+    paramError("gravity", "Gravity should be provided when bouyancy corrections are active.");
```

Comment:
We need this errors for the turbulent prandtl number too I think.

Reply:
Pr_t is already defaulted to 0.9, which is what we used in most turbulent modeling cases. I think is alright to use the default.